src/main/java/org/opensearch/performanceanalyzer/writer/EventLogQueueProcessor.java

```diff
@@ -162,6 +170,33 @@ public void purgeQueueAndPersist() {
             eventLogFileHandler.writeTmpFile(nextMetrics, nextTimeBucket);
         }
         LOG.debug("Writing to disk complete.");
+
+        // Delete the older event log files every filesCleanupPeriod (defaults to 60)
+        // In case files deletion takes longer/fails, we are okay with eventQueue reaching
+        // its max size (100000), post that {@link PerformanceAnalyzerMetrics#emitMetric()}
+        // will emit metric {@link WriterMetrics#METRICS_WRITE_ERROR} and return.
+        cleanup();
```

Comment:
Because the cleanup is called after the files are written, the first time this thread runs, it will delete even the most recently written files because the call to `cleanup` will not have the `lastCleanupTimeBucket` and hence `deleteAllFiles` will be called deleting everything but the `.tmp` files.

Because the full cleanup is just a one time cleanup and happens at the start, let's do it in the constructor itself ?

Reply:
Agreed. This would cause to lose the first timebucket metrics. I would address this as part of separate PR I will have for StatCollector refactoring.